android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -375,96 +375,44 @@ public String specialOSKFontFilename(String filename) {
-      languageID = KMManager.KMDefault_LanguageID;
-      retVal = false;
-    }
 
     String kbKey = String.format("%s_%s", languageID, keyboardID);
-    //if (kbKey.equals(currentKeyboard))
-    //  return false;
-
-    String keyboardName = "";
-    String languageName = "";
-    keyboardVersion = KMManager.getLatestKeyboardFileVersion(getContext(), packageID, keyboardID);
-
-    setKeyboardRoot(packageID);
-    String keyboardPath = makeKeyboardPath(packageID, keyboardID, keyboardVersion);
-
-    String tFont = "''";
-    String oFont = null;
-    HashMap<String, HashMap<String, String>> kbsInfo = LanguageListUtil.getKeyboardsInfo(context);
-    if (kbsInfo != null) {
-      HashMap<String, String> kbInfo = kbsInfo.get(kbKey);
-      if (kbInfo != null) {
-        keyboardName = kbInfo.get(KMManager.KMKey_KeyboardName).replace("'", "\\'");
-        languageName = kbInfo.get(KMManager.KMKey_LanguageName).replace("'", "\\'");
-        txtFont = getFontFilename(kbInfo.get(KMManager.KMKey_Font));
-        oskFont = getFontFilename(kbInfo.get(KMManager.KMKey_OskFont));
-
-        if (!txtFont.isEmpty()) {
-          tFont = makeFontPaths(kbInfo.get(KMManager.KMKey_Font));
-        }
-
-        if (!oskFont.isEmpty()) {
-          oFont = makeFontPaths(kbInfo.get(KMManager.KMKey_Font));
-        } else {
-          oskFont = null;
-        }
-
-        if (oFont == null) {
-          oFont = tFont;
-        }
-      } else {
-        return false;
-      }
-    } else {
-      return false;
+    int index = KeyboardController.getInstance().getKeyboardIndex(kbKey);
+    Keyboard kbInfo = null;
+    if (index != KeyboardController.INDEX_NOT_FOUND) {
+      kbInfo = KeyboardController.getInstance().getKeyboardInfo(index);
     }
 
-    if (tFont.equals("''")) {
-      tFont = fontUndefined;
-    }
-    if (oFont.equals("''")) {
-      oFont = fontUndefined;
-    }
-
-    // Escape single-quoted names for javascript call
-    keyboardName = keyboardName.replaceAll("\'", "\\\\'"); // Double-escaped-backslash b/c regex.
-    languageName = languageName.replaceAll("\'", "\\\\'");
-
-    String jsFormat = "setKeymanLanguage('%s','%s','%s','%s','%s', %s, %s, '%s')";
-    String jsString = String.format(jsFormat, keyboardName, keyboardID, languageName, languageID, keyboardPath, tFont, oFont, packageID);
-    loadJavascript(jsString);
-
-    this.packageID = packageID;
-    this.keyboardID = keyboardID;
-    this.keyboardName = keyboardName;
-    this.keyboardVersion = keyboardVersion;
-    currentKeyboard = kbKey;
-    keyboardSet = true;
-    saveCurrentKeyboardIndex();
-    if (dismissHelpBubble()) {
-      Handler handler = new Handler();
-      handler.postDelayed(new Runnable() {
-        @Override
-        public void run() {
-          loadJavascript("showHelpBubble()");
-        }
-      }, 2000);
+    if (!KMManager.shouldAllowSetKeyboard() || kbInfo == null) {
+      Toast.makeText(context, "Can't set " + packageID + "::" + keyboardID + " for " +
```

Comment:
Should this be reported to sentry as well?

Reply:
Done in a new method `sendError()` that makes the Toast notification and send the message to Sentry.